dbms/src/Storages/DeltaMerge/Filter/IntegerSet.h

```diff
@@ -0,0 +1,292 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Core/Field.h>
+#include <Storages/DeltaMerge/BitmapFilter/BitmapFilter.h>
+#include <Storages/DeltaMerge/Index/InvertedIndex/Reader_fwd.h>
+#include <common/types.h>
+
+#include <memory>
+
+namespace DB::DM
+{
+
+class IntegerSet;
+using IntegerSetPtr = std::shared_ptr<IntegerSet>;
+
+using Fields = std::vector<Field>;
+
+enum class SetType : UInt8
+{
+    Empty,
+    All,
+    Value,
+    Range,
+    Composite,
+};
+
+// IntegerSet is a collection of integer values.
+// It can be multiple values, a range of values, or a composite of multiple sets.
+class IntegerSet : public std::enable_shared_from_this<IntegerSet>
+{
+public:
+    explicit IntegerSet(SetType type_)
+        : type(type_)
+    {}
+
+    virtual ~IntegerSet() = default;
+
+    virtual SetType getType() const = 0;
```

Comment:
```suggestion
    SetType getType() const { return type; }
```
We can either store `type` in `IntegerSet` and define `IntegerSet::getType()` to return type, or make it a pure virtual function and override it in each subclass.